datasets/jigsaw_unintended_bias/jigsaw_unintended_bias.py

```diff
@@ -0,0 +1,156 @@
+                    "sad": datasets.Value("int32"),
+                    "likes": datasets.Value("int32"),
+                    "disagree": datasets.Value("int32"),
+                    "sexual_explicit": datasets.Value("float"),
+                    "identity_annotator_count": datasets.Value("int32"),
+                    "toxicity_annotator_count": datasets.Value("int32"),
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        data_dir = os.path.abspath(os.path.expanduser(dl_manager.manual_dir))
+
+        if not os.path.exists(data_dir):
+            raise FileNotFoundError(
+                "{} does not exist. Make sure you insert a manual dir via `datasets.load_dataset('jigsaw_unintended_bias', data_dir=...)`. Manual download instructions: {}".format(
+                    data_dir, self.manual_download_instructions
+                )
+            )
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"path": os.path.join(data_dir, "train.csv"), "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split("test_private_leaderboard"),
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"path": os.path.join(data_dir, "test_private_expanded.csv"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split("test_public_leaderboard"),
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"path": os.path.join(data_dir, "test_public_expanded.csv"), "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, split: str = "train", path: str = None):
+        """Yields examples."""
+        # This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        data = pd.read_csv(path)
```

Comment:
This will load all the data in memory (1GB), could it be possible to iterate on the csv file line by line maybe ?

Reply:
Done, chunks of 50k now which should be small enough.